### DISCUSSION TITLE
Best practice for distributing [airflow.cfg] over workers?

### DISCUSSION THREAD

**Discussion by ud803** [apache/airflow#17389] [Q&A] [answered]
Hi, I'm newbie to airflow, and I saw the documentation that when using Kubernetes or Celery Executor, it's best 
1) to make DAG files as git repo, and sync it across workers
2) And sync configs [airfow.cfg] across workers as well.

I understand the first point, but how can I best implement number two? It's a config file and I thought it's not a good idea to make it as git repo. What would be the best way to sync configs over main airflow server & workers?

Thanks!

**Comment by potiuk** [comment, marked as answer]
Depending on your deployment. In some cases you can configure deployment in the way that it will automatically put the content on the file for all your "deployments" - this is for example what happen in our helm chart https://airflow.apache.org/docs/helm-chart/stable/index.html  . Another way is to override values and configure them via environment variables (see https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html). Also (see the previous link) some configuration variables (especially the sensitive ones) can be configured via Secret Manager  (when you have _secret added to the configuration variables) - which will make them "automatically" read from single source. 

You can also "bake" your config file in the image or in your scripts that you use to automatically deploy your Airflow (providing that you have such scripts) - if you are using ansible or puppet or terraform or similar ways of 'infrastructure as a code' approach.

Yet another option is to configure Git Sync or Shared Volumes (like NFS/EFS) that are accessed by each of your components (but since configuration values change very rarely I think this is not really good idea. Remember that when you change those configuration values, you have to - in most cases - restart Airflow to take the new values into account. So you might easily end up with changed config but airflow using previous version of the config because it has not been restarted. Embedding configuration together with your "deployment" mechanims seems like much more natural and safe approach.

**Reply by ud803** [reply]
Thanks! I decided to make a config updater with python code, and override airflow.cfg with .env settings!
